src/test/java/org/assertj/core/api/object/ObjectAssert_extracting_with_function_array_Test.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2019 the original author or authors.
+ */
+package org.assertj.core.api.object;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowable;
+import static org.assertj.core.api.BDDAssertions.then;
+import static org.assertj.core.error.ShouldNotBeNull.shouldNotBeNull;
+import static org.assertj.core.presentation.UnicodeRepresentation.UNICODE_REPRESENTATION;
+import static org.assertj.core.test.AlwaysEqualComparator.ALWAY_EQUALS;
+import static org.assertj.core.test.AlwaysEqualComparator.ALWAY_EQUALS_STRING;
+
+import java.util.List;
+import java.util.function.Function;
+
+import org.assertj.core.api.AbstractListAssert;
+import org.assertj.core.api.ObjectAssert;
+import org.assertj.core.internal.Objects;
+import org.assertj.core.test.Employee;
+import org.assertj.core.test.Name;
+import org.assertj.core.util.introspection.PropertyOrFieldSupport;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests for <code>{@link ObjectAssert#extracting(Function[])}</code>.
+ */
+class ObjectAssert_extracting_with_function_array_Test {
+
+  private Employee luke;
+
+  private static final Function<Employee, String> firstName = employee -> employee.getName().getFirst();
+
+  @BeforeEach
+  void setUp() {
+    luke = new Employee(2L, new Name("Luke", "Skywalker"), 26);
+  }
+
+  @Test
+  void should_allow_extracting_values_using_multiple_method_reference_extractors() {
+    // WHEN
+    AbstractListAssert<?, ?, Object, ?> result = assertThat(luke).extracting(Employee::getName, Employee::getAge);
+    // THEN
+    result.hasSize(2)
+          .doesNotContainNull();
+  }
+
+  @Test
+  void should_allow_extracting_values_using_multiple_lambda_extractors() { // FIXME given/when/then
+    // WHEN
+    AbstractListAssert<?, ?, Object, ?> result = assertThat(luke).extracting(firstName, employee -> employee.getName().getLast());
```

Comment:
minor: extract `employee -> employee.getName().getLast()` to `lastName` in a `GIVEN` section